01-import_and_filter.py

```diff
@@ -9,90 +9,102 @@
-        raw_fname_out = op.join(meg_subject_dir,
-                                config.base_fname.format(**locals()))
-
-        print("Input: ", raw_fname_in)
-        print("Output: ", raw_fname_out)
-
-        if not op.exists(raw_fname_in):
-            warn('Run %s not found for subject %s ' %
-                 (raw_fname_in, subject))
-            continue
-
-        raw = mne.io.read_raw_fif(raw_fname_in,
-                                  allow_maxshield=config.allow_maxshield,
-                                  preload=True, verbose='error')
-
-        # add bad channels
-        raw.info['bads'] = bads
-        print("added bads: ", raw.info['bads'])
-
-        if config.set_channel_types is not None:
-            raw.set_channel_types(config.set_channel_types)
-        if config.rename_channels is not None:
-            raw.rename_channels(config.rename_channels)
-
-        # Band-pass the data channels (MEG and EEG)
-        print("Filtering data between %s and %s (Hz)" %
-              (config.l_freq, config.h_freq))
-        raw.filter(
-            config.l_freq, config.h_freq,
-            l_trans_bandwidth=config.l_trans_bandwidth,
-            h_trans_bandwidth=config.h_trans_bandwidth,
-            filter_length='auto', phase='zero', fir_window='hamming',
-            fir_design='firwin')
-
-        if config.resample_sfreq:
-            print("Resampling data to %.1f Hz" % config.resample_sfreq)
-            raw.resample(config.resample_sfreq, npad='auto')
-
-        raw.save(raw_fname_out, overwrite=True)
-        n_raws += 1
-
-        if config.plot:
-            # plot raw data
-            raw.plot(n_channels=50, butterfly=True, group_by='position')
-
-            # plot power spectral densitiy
-            raw.plot_psd(area_mode='range', tmin=10.0, tmax=100.0,
-                         fmin=0., fmax=50., average=True)
-
-    if n_raws == 0:
-        raise ValueError('No input raw data found.')
+    """Filter data from a single subject."""
+    print('\nProcessing subject: {}\n{}'
+          .format(subject, '-' * (20 + len(subject))))
+
+    # Construct the search path for the data file
+    subject_path = op.join('sub-{}'.format(subject), config.kind)
+    if config.ses:
+        subject_path = op.join(subject_path, 'ses-{}'.format(config.ses))
+    data_dir = op.join(config.bids_root, subject_path)
+
+    bids_basename = make_bids_basename(subject=subject,
+                                       session=config.ses,
+                                       task=config.task,
+                                       acquisition=config.acq,
+                                       run=config.run,
+                                       processing=config.proc,
+                                       recording=config.rec,
+                                       space=config.space
+                                       )
+
+    # Find the data file
+    search_str = op.join(data_dir, bids_basename) + '_' + config.kind + '*'
```

Comment:
There is a problem what it searches for. I get:

```py
ValueError: Could not find input data file matching: "/home/mainak/mne_data/ds000117-bids/sub-01/meg/ses-01/sub-01_ses-01_task-VisualFaces_meg*"
```

See how it is `sub-01/meg/ses-01/` instead of `sub-01/ses-01/meg/`

Reply:
I see, thanks for spotting.

Fixed in https://github.com/mne-tools/mne-study-template/pull/35/commits/f11bd945902d291659195f195c17816f926268ed